ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -1316,7 +1282,7 @@ private Directory getAcidState() throws IOException {
     }
 
 
-    private AcidDirInfo callInternal() throws IOException {
+    private AcidDirectory callInternal() throws IOException {
       if (context.acidOperationalProperties != null
```

Comment:
I played around this a bit and I think will do a follow up ticket for mm, because this again is very messy.
For mm tables HiveInputformat will take the partition/table directory, find the delta directories in it and validate them against the validwriteid list. This is neccessary to handle every other file format. But the consequence of this, that the OrcInputFormat#getSplits will be called with the delta directories directly, not with the partition/table dir as it is done normally. So this bypass actually will do a listing in the delta dir itself. And "findOriginals" not only finds the original for MM tables it finds all of the files in the table, and it is necessary otherwise we would not generate split for normal bucketfiles. So the sort answer is that these originals is not the same originals as the context of mm tables ...
The most disgusting part is, when you create an orc table that is not transactional, this method will be called, and we will create a full AcidDirectory just to list all the files in the table as originals. OrcInputformat is very much interweaved with acid code.

Reply:
Thanks for taking the time to research & explain this!